wrapanapi/ec2.py

```diff
@@ -412,6 +412,11 @@ def deploy_template(self, template, *args, **kwargs):
         self.wait_vm_running(instances[0].id, num_sec=timeout)
 
         if vm_name:
+            if self.does_vm_exist(vm_name):
```

Comment:
I suspect this check has to be done before running instance and etc. You won't need to terminate it then.